sound/soc/sof/topology.c

```diff
@@ -3064,7 +3070,7 @@ static int sof_link_hda_load(struct snd_soc_component *scomp, int index,
 	config->hdr.size = size;
 
 	/* get any bespoke DAI tokens */
-	ret = sof_parse_tokens(scomp, config, hda_tokens,
+	ret = sof_parse_tokens(scomp, &config->hda, hda_tokens,
```

Comment:
I wonder if this belongs in a separate patch. This was obviously never right, so maybe recognize it as a fix, then add new things?

Reply:
> I wonder if this belongs in a separate patch. This was obviously never right, so maybe recognize it as a fix, then add new things?

Done